src/core/sync/mutex.d

```diff
@@ -55,31 +57,45 @@ class Mutex :
     /**
      * Initializes a mutex object.
      *
-     * Throws:
-     *  SyncError on error.
      */
-    this() nothrow @trusted
+    this() @trusted nothrow @nogc
+    {
+        this(true);
+    }
+
+    /// ditto
+    this() shared @trusted nothrow @nogc
     {
-        version( Windows )
+        this(true);
+    }
+
+    // Undocumented, useful only in Mutex.this().
+    private this(this Q)(bool _unused_) @trusted nothrow @nogc
+        if (is(Q == Mutex) || is(Q == shared Mutex))
+    {
+        version (Windows)
         {
-            InitializeCriticalSection( &m_hndl );
+            InitializeCriticalSection(cast(CRITICAL_SECTION*)&m_hndl);
         }
-        else version( Posix )
+        else version (Posix)
         {
+            import core.internal.abort : abort;
             pthread_mutexattr_t attr = void;
 
-            if( pthread_mutexattr_init( &attr ) )
-                throw new SyncError( "Unable to initialize mutex" );
-            scope(exit) pthread_mutexattr_destroy( &attr );
+            !pthread_mutexattr_init(&attr) ||
+                abort("Unable to initialize mutex");
 
-            if( pthread_mutexattr_settype( &attr, PTHREAD_MUTEX_RECURSIVE ) )
-                throw new SyncError( "Unable to initialize mutex" );
+            scope (exit) pthread_mutexattr_destroy(&attr);
```

Comment:
It seems inconsistent that the result of `pthread_mutexattr_destroy` is not checked. It probably should be. I'm not sure that that's really blocker though, since it hasn't been being checked. It would just be an improvement that was missed with this PR.